src/display/annotation_layer.js

```diff
@@ -210,41 +214,13 @@ class AnnotationElement {
       page.view[3] - data.rect[3] + page.view[1],
     ]);
 
-    if (data.hasOwnCanvas) {
-      const transform = viewport.transform.slice();
-      const [scaleX, scaleY] = Util.singularValueDecompose2dScale(transform);
-      width = Math.ceil(width * scaleX);
-      height = Math.ceil(height * scaleY);
-      rect[0] *= scaleX;
-      rect[1] *= scaleY;
-      // Reset the scale part of the transform matrix (which must be diagonal
-      // or anti-diagonal) in order to avoid to rescale the canvas.
-      // The canvas for the annotation is correctly scaled when it is drawn
-      // (see `beginAnnotation` in canvas.js).
-      for (let i = 0; i < 4; i++) {
-        transform[i] = Math.sign(transform[i]);
-      }
-      container.style.transform = `matrix(${transform.join(",")})`;
-    } else {
-      container.style.transform = `matrix(${viewport.transform.join(",")})`;
-    }
-
-    container.style.transformOrigin = `${-rect[0]}px ${-rect[1]}px`;
-
     if (!ignoreBorder && data.borderStyle.width > 0) {
-      container.style.borderWidth = `${data.borderStyle.width}px`;
-      if (data.borderStyle.style !== AnnotationBorderStyleType.UNDERLINE) {
-        // Underline styles only have a bottom border, so we do not need
-        // to adjust for all borders. This yields a similar result as
-        // Adobe Acrobat/Reader.
-        width -= 2 * data.borderStyle.width;
-        height -= 2 * data.borderStyle.width;
-      }
+      container.style.borderWidth = ` ${data.borderStyle.width}px`;
 
       const horizontalRadius = data.borderStyle.horizontalCornerRadius;
       const verticalRadius = data.borderStyle.verticalCornerRadius;
       if (horizontalRadius > 0 || verticalRadius > 0) {
-        const radius = `${horizontalRadius}px / ${verticalRadius}px`;
+        const radius = `calc(${horizontalRadius}px * var(--scale-factor)) / calc(${verticalRadius}px * var(--scale-factor))`;
```

Comment:
Why do we need to multiply with `var(--scale-factor)` in both the numerator *and* denominator, since I'd expect that to cancel out?

Reply:
I don't think the slash is used for a division:
https://developer.mozilla.org/en-US/docs/Web/CSS/border-radius#syntax
but  it's only a way to define two radius in order to have an elliptic curvature.
But maybe I overlooked something...

Reply:
Sorry, I got lost in the parenthesis (since Prettier doesn't required a line-break here).